sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -897,43 +921,41 @@ STRING
     ;
 
 BIGINT_LITERAL
-    : DIGIT+ 'L'
+    : DIGIT+ 'L' {isValidNumber()}?
     ;
 
 SMALLINT_LITERAL
-    : DIGIT+ 'S'
+    : DIGIT+ 'S' {isValidNumber()}?
     ;
 
 TINYINT_LITERAL
-    : DIGIT+ 'Y'
+    : DIGIT+ 'Y' {isValidNumber()}?
     ;
 
 BYTELENGTH_LITERAL
-    : DIGIT+ ('B' | 'K' | 'M' | 'G')
+    : DIGIT+ ('B' | 'K' | 'M' | 'G') {isValidNumber()}?
     ;
 
 INTEGER_VALUE
-    : DIGIT+
+    : DIGIT+ {isValidNumber()}?
     ;
 
 DECIMAL_VALUE
-    : DIGIT+ '.' DIGIT*
-    | '.' DIGIT+
+    : DECIMAL_DIGITS {isValidNumber()}?
```

Comment:
SCIENTIFIC_DECIMAL_VALUE, DOUBLE_LITERAL, and BIGDECIMAL_LITERAL may also cause this issue.